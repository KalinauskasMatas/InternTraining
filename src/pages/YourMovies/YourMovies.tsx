import NavBar from "../../components/NavBar/NavBar";

const YourMovies = () => {
  return (
    <div className="body-wrapper">
      <NavBar />
      Your movies
    </div>
  );
};

export default YourMovies;
