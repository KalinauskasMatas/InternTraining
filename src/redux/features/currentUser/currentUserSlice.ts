import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "../../store";
import { UserState } from "../../interfaces";

const initialState: UserState =
  JSON.parse(localStorage.getItem("currUser") as string) || {};

const currentUserSlice = createSlice({
  name: "currentUser",
  initialState,
  reducers: {
    setUser: (state, action: PayloadAction<UserState>) => {
      state.fname = action.payload.fname;
      state.surname = action.payload.surname;
      state.email = action.payload.email;
      state.password = action.payload.password;
      localStorage.setItem("currUser", JSON.stringify(state));
    },
    clearUser: (state) => {
      localStorage.removeItem("currUser");
      state = { ...initialState };
      console.log(state);
    },
  },
});

export const { setUser, clearUser } = currentUserSlice.actions;

export const selectCurrUser = (state: RootState) => state.currentUser;

export default currentUserSlice.reducer;
