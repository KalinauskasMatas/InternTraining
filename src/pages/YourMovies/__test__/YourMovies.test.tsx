import YourMovies from "../YourMovies";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import currentUserSlice from "../../../redux/features/currentUser/currentUserSlice";

import { BrowserRouter } from "react-router-dom";

afterAll(cleanup);

const store = configureStore({
  reducer: {
    currentUser: currentUserSlice,
  },
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <YourMovies />
      </BrowserRouter>
    </Provider>
  );
});

it("matches with snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <YourMovies />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
