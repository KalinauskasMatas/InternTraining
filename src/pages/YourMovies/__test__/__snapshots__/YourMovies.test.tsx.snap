// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with snapshot 1`] = `
<div
  className="body-wrapper"
>
  <header
    className="header-name"
    data-testid="header-title"
  >
    Movie rental
  </header>
  <nav>
    <ul>
      <span
        className="left"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/yourMovies"
            onClick={[Function]}
          >
            Your movies
          </a>
        </li>
        <li
          className="profile-link"
        >
          <a
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
      </span>
      <span>
        <li
          id="logout"
          onClick={[Function]}
        >
          Logout
        </li>
      </span>
    </ul>
  </nav>
  <div
    className="body-wrapper"
  >
    <section
      className="your-movies"
    >
      <h2
        className="your-movies-header"
      >
        Your movies
      </h2>
      <table
        id="your-movies-table"
      >
        <tbody>
          <tr>
            <th>
              Name
            </th>
            <th>
              Genre
            </th>
            <th>
              Time
            </th>
            <th>
              Price
            </th>
            <th />
          </tr>
        </tbody>
      </table>
    </section>
  </div>
  <footer>
    <span>
      Made by: Matas Kalinauskas
    </span>
    <span>
      2023
    </span>
  </footer>
</div>
`;
